src/main/java/org/killbill/billing/plugin/adyen/api/AdyenPaymentTransactionInfoPlugin.java

```diff
@@ -156,7 +139,33 @@ public PaymentPluginStatus getStatus() {
         }
     }
 
-    private static PaymentPluginStatus getPaymentPluginStatus(final PaymentServiceProviderResult pspResult) {
+    /**
+     * Transforms adyenCallErrorStatus (where there any technical errors?) and pspResult (was the call successful from a business perspective) into the PaymentPluginStatus.
+     * Therefor
+     */
+    private static PaymentPluginStatus getPaymentPluginStatus(final Optional<AdyenCallErrorStatus> adyenCallErrorStatus, final Optional<PaymentServiceProviderResult> pspResult) {
+        checkArgument(adyenCallErrorStatus.isPresent() ^ pspResult.isPresent());
+        return (pspResult.isPresent()) ? pspResultToPaymentPluginStatus(pspResult.get()) : adyenCallErrorStatusToPaymentPluginStatus(adyenCallErrorStatus.get());
+    }
+
+    private static PaymentPluginStatus adyenCallErrorStatusToPaymentPluginStatus(final AdyenCallErrorStatus adyenCallErrorStatus) {
+        switch (adyenCallErrorStatus) {
+            case REQUEST_NOT_SEND:
+                return PaymentPluginStatus.CANCELED;
+            case RESPONSE_ABOUT_INVALID_REQUEST:
+                return PaymentPluginStatus.ERROR;
+            case RESPONSE_NOT_RECEIVED:
```

Comment:
Why did you break down the `UNDEFINED` cases? You could simplify `mapExceptionToCallResult` below by just looking for `CANCELED` and `ERROR` exceptions.